apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java

```diff
@@ -0,0 +1,360 @@
+      } finally {
+        recursivelyDelete(tempDir);
+      }
+    }
+  }
+
+  /**
+   * Extracts a zip file specified by the zipFilePath to a directory specified by destDirectory
+   * (will be created if does not exists).
+   *
+   * @param zipFilePath the zip file path
+   * @param destDirectory the destination directory
+   * @throws IOException (any errors encountered will be bubbled up)
+   */
+  public static void unzip(Path zipFilePath, Path destDirectory) throws IOException {
+    ZipInputStream zipIn =
+        new ZipInputStream(new FileInputStream(zipFilePath.toFile().getAbsolutePath()));
+    ZipEntry entry = zipIn.getNextEntry();
+    // iterates over entries in the zip file
+    while (entry != null) {
+      Path filePath = Paths.get(destDirectory.toFile().getAbsolutePath(), entry.getName());
+      if (!entry.isDirectory()) {
+        // if the entry is a file, extracts it
+        extractFile(zipIn, filePath);
+      } else {
+        // if the entry is a directory, make the directory
+        File dir = new File(filePath.toFile().getAbsolutePath());
+        dir.mkdir();
+      }
+      zipIn.closeEntry();
+      entry = zipIn.getNextEntry();
+    }
+    zipIn.close();
+  }
+
+  /**
+   * Extracts a zip entry (file entry).
+   *
+   * @param zipIn the zip file coming in
+   * @param filePath the file path for the file
+   * @throws IOException (any errors encountered will be bubbled up)
+   */
+  public static void extractFile(ZipInputStream zipIn, Path filePath) throws IOException {
+    BufferedOutputStream bos =
+        new BufferedOutputStream(new FileOutputStream(filePath.toFile().getAbsolutePath()));
+    byte[] bytesIn = new byte[BUFFER_SIZE];
+    int read = 0;
+    while ((read = zipIn.read(bytesIn)) != -1) {
+      bos.write(bytesIn, 0, read);
+    }
+    bos.close();
+  }
+
+  /**
+   * Creates the file in the specified location.
+   *
+   * @param workingDir a directory that temporary/working files can be written to
+   * @param fileName the output file/resource to produce
+   * @throws IOException (any errors encountered will be bubbled up)
+   * @return path to file
+   */
+  public static Path getOriginalNpiDataFile(Path workingDir, String fileName)
+      throws IOException, IllegalStateException {
+    // download NPI file
+    Path downloadedNpiZipFile =
+        Paths.get(workingDir.resolve("npidata.zip").toFile().getAbsolutePath());
+    URL ndctextZipUrl = new URL(fileName);
+    if (!Files.isReadable(downloadedNpiZipFile)) {
+      // connectionTimeout, readTimeout = 10 seconds
+      final Integer connectionTimeout = 100000;
+      final Integer readTimeout = 100000;
+
+      FileUtils.copyURLToFile(
+          ndctextZipUrl,
+          new File(downloadedNpiZipFile.toFile().getAbsolutePath()),
+          connectionTimeout,
+          readTimeout);
+    }
+
+    // unzip NPI file.  Zip file contains these files
+    // pl_pfile_20050523-20220410.csv
```

Comment:
I'm guessing that a couple of these filenames are predicated on date (from yyyymmdd-yyyymmdd) so may not be exactly as comment says.